ovs/lib/alba.py

```diff
@@ -1114,260 +1115,287 @@ def _validate_asd_removal(storage_router_ip):
                                                               number=nsm_cluster.number)
-                                AlbaController._logger.debug('  Restart sequence')
+                                AlbaController._logger.debug('ALBA Backend {0} - Restarting cluster'.format(alba_backend.name))
                                 arakoon_installer.restart_cluster_after_extending(new_ip=candidate_sr.ip)
                                 AlbaController._update_nsm(abm_name=abm_cluster_name,
                                                            nsm_name=nsm_cluster.name,
                                                            ip=candidate_sr.ip)
-                                AlbaController._logger.debug('Node added')
-
-                # Load and minimum nsm hosts
-                nsms_to_add = additional_nsm_amount
-                load_ok = min(nsm_loads.values()) < maxload
-                AlbaController._logger.debug('Currently {0} NSM hosts'.format(len(nsm_loads)))
-                if min_nsms > 1:
-                    AlbaController._logger.debug('Minimum {0} NSM hosts requested'.format(min_nsms))
-                    nsms_to_add = max(0, min_nsms - len(nsm_loads))
-                if load_ok:
-                    AlbaController._logger.debug('NSM load OK')
-                else:
-                    AlbaController._logger.debug('NSM load NOT OK')
-                    nsms_to_add = max(1, nsms_to_add)
-                if nsms_to_add > 0:
-                    AlbaController._logger.debug('Trying to add {0} NSM hosts'.format(nsms_to_add))
+                                AlbaController._logger.debug('ALBA Backend {0} - Extended cluster'.format(alba_backend.name))
 
-                # Deploy new NSM clusters
+                overloaded = min(nsm_loads.values()) >= maxload
+                if overloaded is False:  # At least 1 NSM is not overloaded yet
+                    AlbaController._logger.debug('ALBA Backend {0} - NSM load OK'.format(alba_backend.name))
+                    if internal is True:
+                        nsms_to_add = max(0, min_nsms - len(nsm_loads))  # When load is not OK, deploy at least 1 additional NSM
+                    else:
+                        nsms_to_add = len(nsm_cluster_names)  # For externally managed clusters we only claim the specified clusters, if none provided, we just log it
+                    if nsms_to_add == 0:
+                        continue
+                else:
+                    AlbaController._logger.warning('ALBA Backend {0} - NSM load is NOT OK'.format(alba_backend.name))
+                    if internal is True:
+                        nsms_to_add = max(1, min_nsms - len(nsm_loads))  # When load is not OK, deploy at least 1 additional NSM
+                    else:
+                        nsms_to_add = len(nsm_cluster_names)  # For externally managed clusters we only claim the specified clusters, if none provided, we just log it
```

Comment:
Shouldn't the code check if the supplied external nsms are sufficient to add? Might be better to lo gsuch a thing
So if nsms_to_add  < len(nsm_cluster_names): log something about not having enough external nsms supplied to satisfy the current safety

Reply:
These are the clusters that will be claimed by the checkup. At this point they have been verified to be available AND not in use